cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeletedProgramHandlerStage.java

```diff
@@ -125,39 +112,16 @@ public void process(ApplicationDeployable appSpec) throws Exception {
     emit(appSpec);
   }
 
-  private void deleteMetrics(String account, String application, Iterable<String> flows) throws IOException {
-    ServiceDiscovered discovered = discoveryServiceClient.discover(Constants.Service.METRICS);
-    Discoverable discoverable = new RandomEndpointStrategy(discovered).pick(DISCOVERY_TIMEOUT_SECONDS,
-                                                                            TimeUnit.SECONDS);
-
-    if (discoverable == null) {
-      LOG.error("Fail to get any metrics endpoint for deleting metrics.");
-      return;
-    }
-
+  private void deleteMetrics(String namespace, String application, Iterable<String> flows) throws Exception {
     LOG.debug("Deleting metrics for application {}", application);
     for (String flow : flows) {
-      // TODO: use MetricStore directly to delete the metrics [CDAP-2163]
-      String url = String.format("http://%s:%d%s/metrics/%s/apps/%s/flows/%s",
-                                 discoverable.getSocketAddress().getHostName(),
-                                 discoverable.getSocketAddress().getPort(),
-                                 Constants.Gateway.API_VERSION_3,
-                                 "ignored",
-                                 application, flow);
-
-      SimpleAsyncHttpClient client = new SimpleAsyncHttpClient.Builder()
-        .setUrl(url)
-        .setRequestTimeoutInMs((int) METRICS_SERVER_RESPONSE_TIMEOUT)
-        .build();
-
-      try {
-        client.delete().get(METRICS_SERVER_RESPONSE_TIMEOUT, TimeUnit.MILLISECONDS);
-      } catch (Exception e) {
-        LOG.error("exception making metrics delete call", e);
-        Throwables.propagate(e);
-      } finally {
-        client.close();
-      }
+      long endTs = System.currentTimeMillis() / 1000;
+      Map<String, String> tags = Maps.newHashMap();
+      tags.put(Constants.Metrics.Tag.NAMESPACE, namespace);
+      tags.put(Constants.Metrics.Tag.APP, application);
+      tags.put(Constants.Metrics.Tag.FLOW, flow);
+      MetricDeleteQuery deleteQuery = new MetricDeleteQuery(0, endTs, null, tags);
```

Comment:
seems like we should have ctor without metric name param, if null is allowed - would make this code pretty. Nothing to do in this PR, just noting - for future